cpp/dolfin/fem/dofs_permutation.cpp

```diff
@@ -429,233 +429,152 @@ hexahedron_rotations_and_reflection(const int volume_dofs, const int blocksize)
 }
 //-----------------------------------------------------------------------------
 Eigen::Array<std::int8_t, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>
-compute_ordering_triangle(const mesh::Mesh& mesh)
+compute_ordering(const mesh::Mesh& mesh)
 {
-  const int num_cells = mesh.num_entities(mesh.topology().dim());
-  const int num_permutations = get_num_permutations(mesh.cell_type());
-  Eigen::Array<std::int8_t, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>
-      cell_orders(num_cells, num_permutations);
-
-  const std::vector<std::int64_t>& global_indices
-      = mesh.topology().global_indices(0);
-  // Set orders for each cell
-  for (int cell_n = 0; cell_n < num_cells; ++cell_n)
-  {
-    const mesh::MeshEntity cell(mesh, 2, cell_n);
-    const int* vertices = cell.entities(0);
-    const std::int64_t v0 = global_indices[vertices[0]];
-    const std::int64_t v1 = global_indices[vertices[1]];
-    const std::int64_t v2 = global_indices[vertices[2]];
-
-    // Set the orders for the edge flips
-    cell_orders(cell_n, 0) = (v1 > v2);
-    cell_orders(cell_n, 1) = (v0 > v2);
-    cell_orders(cell_n, 2) = (v0 > v1);
-
-    // Set the orders for the face rotation and reflection
-    const std::array<std::int8_t, 2> tri_orders
-        = calculate_triangle_orders<std::int64_t>(v0, v1, v2);
-    cell_orders(cell_n, 3) = tri_orders[0];
-    cell_orders(cell_n, 4) = tri_orders[1];
+  mesh::CellType type = mesh.cell_type();
+  const int t_dim = mesh.topology().dim();
+  const int num_cells = mesh.num_entities(t_dim);
+  const int num_permutations = get_num_permutations(type);
+  const int num_vertices_per_cell = mesh::num_cell_vertices(type);
+
+  // Find the total number of edges + faces + volumes
+  int entity_count = 0;
+  for (int dim = 1; dim < t_dim; ++dim)
+    entity_count += mesh::cell_num_entities(type, dim);
+  entity_count += 1;
+
+  // Get lists of vertices on each entity from the cell type
+  std::vector<std::pair<mesh::CellType, Eigen::Array<int, 1, Eigen::Dynamic>>>
+      entities(entity_count);
+  { // scope for j
+    int j = 0;
+    for (int dim = 1; dim < t_dim; ++dim)
+    {
+      auto vertices = mesh::get_entity_vertices(type, dim);
```

Comment:
Best to replace most of the `auto` to improve readability.

Reply:
I've replaced most, but left a couple in where the type is very long so I think would hurt readability (eg `std::function<Eigen::Array<std::int8_t, 1, Eigen::Dynamic>Eigen::Array<std::size32_t, 1, Eigen::Dynamic>)>`)